best-practices/puppet_prs.md

```diff
@@ -0,0 +1,36 @@
+# Puppet PRs
+
+Occasionally, a developer may need a system level configuration change made to a machine. Perhaps the most common example is when a dev needs ssh access to a machine to be able to kick off a capistrano task. In this case, we encourage devs to make a pull request to the Puppet repository where such configuration is mannaged.
+
+To issue a PR to the DLSS puppet repository, first make sure you have a local clone of the repo. Inside, check out a branch off of `production` on which to do your work. By convention the name of the branch should include your sunetid, so that operations knows who is issuing the PR:
```

Comment:
Is the branch name convention something used by ops? Is this redundant to the GitHub user issuing the PR? I'm curious about how this convention is used.

Reply:
Good question. On reflection, I don't think "so that operations knows who is issuing the PR" is quite to the point. I'll try again: the way we test out puppet changes often involves running puppet directly on a machine against a particular branch. When we do that, there's a `puppet.conf` file whose `environment` setting will be set to the branch name.  If that branch name includes a sunetid, it gives ops some visibility right there into who has been working on the machine.